core/src/test/java/org/hibernate/ogm/utils/OgmTestRunner.java

```diff
@@ -140,6 +140,8 @@ protected void runChild(FrameworkMethod method, RunNotifier notifier) {
 		}
 		finally {
 			if ( testMethodScopedSessionFactory != null ) {
+				cleanUpPendingTransactionIfRequired();
+				TestHelper.dropSchemaAndDatabase( testScopedSessionFactory );
 				testMethodScopedSessionFactory.close();
```

Comment:
Yes, it's a tough choice. Last time I extracted a lot of small changes like that, people complained about it because it was hard to see what was the overall purpose of all these changes. So, for this one, I extracted the bugs I found in separate issues and kept the rest in it. Moreover extracting this sort of things adds quite a lot of overhead to the work too. Anyway, I agree with you, it's not easy to find the right balance and I might be on the bad side on this one.